modules/client/examples/01-client/16-is-dao-update-proposal-valid.ts

```diff
@@ -0,0 +1,37 @@
+/* MARKDOWN
+---
+title: Check if a proposal valid for updating a dao
+---
+
+### Check if a proposal valid for updating a dao
+
+It will check that the proposal has the correct actions to update a dao.
+It will check that the eencoded previous version matches the dao version
+it will check that the implementation address matches the latest dao factory address
+It will check that the init data is empty becuase the dao factory does not have new params so init data is not necessary
```

Comment:
Mmm, from what I saw on the figma this should be more or less generic, not specific for the version 1.3.0. Atm this would only work for the latest version of the contracts, this means if we are in version 1.0.0 and want to upgrade to 1.3.0 but 1.4.0 is alredy out this will return an error on the implementation address because it will compare it to the latest one, and we dont have a way of fixing this at the moment because the active contracts only hold the latest version

Reply:
We solved this in the version you reviewed we are still missing a way of knowing if the update will need init data